### ISSUE TITLE
Inside emacs, SC2148 seems to be overruled 

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1179]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2148
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
# -*- sh -*-

```

#### Here's what shellcheck currently says:
```
Line 1	SC2148: Tips depend on target shell and yours is unknown. Add a shebang.
```
#### Here's what I wanted or expected to see:

When inside an editor (to wit, emacs) I do not ever see this error when I do not have a shebang. It guess it is an editor integration thing; is the editor feeding the shell type to shellcheck? I don't think it should.

**Comment by Nightfirecat** [comment]
From [SC2148's wiki page](/koalaman/whellcheck/wiki/sc2148):

> ### Rationale:
> 
> Different shells support different features. To give effective advice, ShellCheck needs to know which shell your script is going to run on. You will get a different numbers of warnings about different things depending on your target shell.
> 
> ShellCheck normally determines your target shell from the shebang (having e.g. `#!/bin/sh` as the first line). The shell can also be specified from the CLI with `-s`, e.g. `shellcheck -s sh file`.
> 
> If you don't specify shebang nor `-s`, ShellCheck gives this message and proceeds with some default (`bash`).
> 
> Note that this error can not be ignored with a [[directive]]. It is not a suggestion to improve your script, but a warning that ShellCheck lacks information it needs to be helpful.

**Comment by charlescanato** [comment]
Matthew, this *is* the case. I had my own quota of problems with Flycheck on Emacs, trying to make it use only Shellcheck (it'd use Shellcheck in addition to Bash checking, by default). But Flycheck *always* uses `-s` (even if it's not what you want).

**Comment by matthewpersico** [comment]
Sounds like it’s time to dig into some lisp code and turn that off. :-). Since this is not a shellcheck issue, I’m going to close it. Thank you all for your help.

**Comment by matthewpersico** [comment]
FYI, as of flycheck-20180326.836/flycheck.el

```lisp
(flycheck-define-checker sh-shellcheck
  "A shell script syntax and style checker using Shellcheck.

See URL `https://github.com/koalaman/shellcheck/'."
  :command ("shellcheck"
            "--format" "checkstyle"
            "--shell" (eval (symbol-name sh-shell))
            (option-flag "--external-sources" flycheck-shellcheck-follow-sources)
            (option "--exclude" flycheck-shellcheck-excluded-warnings list
                    flycheck-option-comma-separated-list)
            "-")
  :standard-input t
  :error-parser flycheck-parse-checkstyle
  :error-filter
  (lambda (errors)
    (flycheck-remove-error-file-names
     "-" (flycheck-dequalify-error-ids errors)))
  :modes sh-mode
  :predicate (lambda () (memq sh-shell flycheck-shellcheck-supported-shells))
  :verify (lambda (_)
            (let ((supports-shell (memq sh-shell
                                        flycheck-shellcheck-supported-shells)))
              (list
               (flycheck-verification-result-new
                :label (format "Shell %s supported" sh-shell)
                :message (if supports-shell "yes" "no")
                :face (if supports-shell 'success '(bold warning)))))))
```

So you would need to remove/override the 

```lisp
"--shell" (eval (symbol-name sh-shell))
```
line to force SC2148 in flycheck mode.
